File: MyInsight/MyInsight/UIFunction/Function/Other/OtherVC.m
//
//  OtherVC.m
//  MyInsight
//
//  Created by SongMenglong on 2018/1/24.
//  Copyright © 2018年 SongMenglong. All rights reserved.
//

#import "OtherVC.h"
#import <SWRevealViewController.h>
#import "ChatListVC.h"
#import "XinHuaVC.h"
#import "ZhiHuDailyVC.h"
#import "CrossVC.h"
#import "ReferToVC.h"
#import "UMengVC.h"

@interface OtherVC ()<UITableViewDelegate, UITableViewDataSource>
// 列表
@property (nonatomic, strong) UITableView *tableView;
// 数组数据
@property (nonatomic, strong) NSArray *dataArray;

@end

static const NSString *BaiSiBuDeJieStr = @"百思不得姐";
static const NSString *QiuShiBaiKeStr = @"糗事百科";
static const NSString *MeiZhiStr = @"妹纸";
static const NSString *ChatUIStr = @"聊天界面";
static const NSString *XinHuaStr = @"中华新华字典";
static const NSString *ZhiHuDailyStr = @"知乎日报";
static const NSString *CrossStr = @"跨平台";
static const NSString *ReferToStr = @"鸣谢:借鉴参考的仓库";
static const NSString *GitHubStr = @"GitHub";
static const NSString *UMengStr = @"友盟UMeng";

@implementation OtherVC

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    // 
    if ([self revealViewController] != NULL) {
        [[self revealViewController] tapGestureRecognizer];
        [self.view addGestureRecognizer:[self revealViewController].panGestureRecognizer];
    }
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    
    if ([self revealViewController] != NULL) {
        [self.view removeGestureRecognizer:[self revealViewController].panGestureRecognizer];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // 左右button
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"左边" style:UIBarButtonItemStylePlain target:[self revealViewController] action:@selector(revealToggle:)];
    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"右边" style:UIBarButtonItemStylePlain target:[self revealViewController] action:@selector(rightRevealToggle:)];
    
    //处理数据
    [self handleTableViewData];
    // 创建列表
    [self creatTableView];
}

// 处理数据
- (void)handleTableViewData {
    self.dataArray = @[ReferToStr, BaiSiBuDeJieStr, QiuShiBaiKeStr, MeiZhiStr, ZhiHuDailyStr, ChatUIStr, XinHuaStr, CrossStr, UMengStr, GitHubStr];
}

#pragma mark - 创建TableView
- (void)creatTableView {
    self.tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
    [self.view addSubview:self.tableView];
    // 设置代理
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
    // 清空多余cell
    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
    // 注册cell
    //[self.tableView registerNib:[UINib nibWithNibName:@"MineCell" bundle:nil] forCellReuseIdentifier:@"MineCell"];
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"UITableViewCell"];
}

#pragma mark - 实现TableView的代理协议
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

// 行数
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataArray.count;
}

// 生成cell
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"UITableViewCell" forIndexPath:indexPath];
    if (cell == nil) {
        //cell = [[[NSBundle mainBundle]loadNibNamed:@"MineCell" owner:self options:nil] lastObject];
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"UITableViewCell"];
    }
    
    // 赋值
    cell.textLabel.text = self.dataArray[indexPath.row];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    
    return cell;
}

// 选中cell
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    // 获取cell的字符串
    NSString *cellString = [self.dataArray objectAtIndex:indexPath.row];
    
    if ([cellString isEqual:ReferToStr]) {
        // 借鉴参考的仓库链接
        ReferToVC *referToVC = [[ReferToVC alloc] init];
        referToVC.hidesBottomBarWhenPushed = YES;
        [self.navigationController pushViewController:referToVC animated:YES];
    }
    if ([cellString isEqual:ChatUIStr]) {
        ChatListVC *chatListVC = [[ChatListVC alloc] init];
        chatListVC.hidesBottomBarWhenPushed = YES;
        [self.navigationController pushViewController:chatListVC animated:YES];
    }
    if ([cellString isEqual:XinHuaStr]) {
        // 中华新华字典
        XinHuaVC *xinHuaVC = [[XinHuaVC alloc] init];
        xinHuaVC.hidesBottomBarWhenPushed = YES;
        [self.navigationController pushViewController:xinHuaVC animated:YES];
    }
    if ([cellString isEqual:ZhiHuDailyStr]) {
        // 知乎日报
        ZhiHuDailyVC *zhiHuDailyVC = [[ZhiHuDailyVC alloc] init];
        zhiHuDailyVC.hidesBottomBarWhenPushed = YES;
        [self.navigationController pushViewController:zhiHuDailyVC animated:YES];
    }
    if ([cellString isEqual:CrossStr]) {
        // 知乎日报
        CrossVC *crossVC = [[CrossVC alloc] init];
        crossVC.hidesBottomBarWhenPushed = YES;
        [self.navigationController pushViewController:crossVC animated:YES];
    }
    if ([cellString isEqual:UMengStr]) {
        // 友盟
        UMengVC *umengVC = [[UMengVC alloc] init];
        umengVC.title = cellString;
        umengVC.hidesBottomBarWhenPushed = YES;
        [self.navigationController pushViewController:umengVC animated:YES];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: MyInsight_Swift/MyInsight_Swift/UIFunction/Function/Other/OtherVC.swift
//
//  OtherVC.swift
//  MyInsight_Swift
//
//  Created by SongMengLong on 2018/7/22.
//  Copyright © 2018年 SongMengLong. All rights reserved.
//

import UIKit
import SWRevealViewController
import SnapKit

class OtherVC: BaseVC {
    // 声明变量
    let tableview = UITableView()
    // 数组
    var dataArray = Array<String>()
    
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        // 抽屉
        if self.revealViewController() != nil {
            self.revealViewController().tapGestureRecognizer()
            // 添加手势动作
            self.view.addGestureRecognizer(self.revealViewController().panGestureRecognizer())
        }
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        
        if self.revealViewController() != nil {
            // 移除掉手势动作
            self.view.removeGestureRecognizer(self.revealViewController().panGestureRecognizer())
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()        
        // 左右button
        self.navigationItem.leftBarButtonItem = UIBarButtonItem.init(title: "左边", style: UIBarButtonItem.Style.plain, target: self.revealViewController(), action: #selector(SWRevealViewController.revealToggle(_:))) //revealToggle(_:)
        self.navigationItem.rightBarButtonItem = UIBarButtonItem.init(title: "右边", style: UIBarButtonItem.Style.plain, target: self.revealViewController(), action: #selector(SWRevealViewController.rightRevealToggle(_:)))
        
        self.title = "其他"
        // 设置背景颜色
        self.view.backgroundColor = UIColor.white
        // 初始化tableview
        
        self.view.addSubview(self.tableview)
        self.tableview.frame = self.view.bounds;
        self.tableview.delegate = self
        self.tableview.dataSource = self
        // 注册cell
        self.tableview.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: "cell")
        // 清空多余cell
        self.tableview.tableFooterView = UIView(frame: CGRect.zero)
        
        dataArray = ["博客", "我的博客"]
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}

//MARK: - 扩展
extension OtherVC: UITableViewDelegate, UITableViewDataSource {
    // 实现代理协议
    
    // setction个数
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    // cell行数目
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataArray.count
    }
    // 生成cell
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = UITableViewCell(style: UITableViewCell.CellStyle.default, reuseIdentifier: "cell")
        // label赋值
        cell.textLabel?.text = dataArray[indexPath.row]
        // 箭头
        cell.accessoryType = UITableViewCell.AccessoryType.disclosureIndicator
        
        return cell
    }
    // 选中cell
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        let cellStr: String = self.dataArray[indexPath.row]
        
        if cellStr == "博客" {
            debugPrint("博客")
            let blogListVC :BlogListVC = BlogListVC()
            blogListVC.hidesBottomBarWhenPushed = true
            self.navigationController?.pushViewController(blogListVC, animated: true)
        }
        
        if cellStr == "我的博客" {
            debugPrint("我的博客")
            let myBlogVC :MyBlogVC = MyBlogVC()
            myBlogVC.hidesBottomBarWhenPushed = true
            self.navigationController?.pushViewController(myBlogVC, animated: true)
        }
    }
}

